main.js

```diff
@@ -1,105 +1,154 @@
+};
+
+let selectedDateRes = selectedDate(daysArray, 6, 13);
+console.log(selectedDateRes)
+
+const days = [{}];
+const selected = [{}];
+const current = [{}];
+
+daysArray.forEach((item) => days.push({ dayOfMonth: item }));
+selectedDateRes.forEach((item) => selected.push({ selectedDay: item }));
+notCurrentMonthRes.forEach((item) => current.push({ notCurrentMonth: item }));
+
+
+const calendar = days.map((item, index) => ({
+  ...item,
+  ...selected[index],
+  ...current[index],
+}));
+console.log(calendar);
+
+console.log("----LESSON-8----");
+const studentsData = [
+  {
+    firstName: "Василий",
+    lastName: "Петров",
+    admissionYear: 2019,
+    courseName: "Java",
+  },
+  {
+    firstName: "Иван",
+    lastName: "Иванов",
+    admissionYear: 2018,
+    courseName: "JavaScript",
+  },
+  {
+    firstName: "Александр",
+    lastName: "Федоров",
+    admissionYear: 2017,
+    courseName: "Python",
+  },
+  {
+    firstName: "Николай",
+    lastName: "Петров",
+    admissionYear: 2019,
+    courseName: "Android",
+  },
+];
+
+class User {
+  constructor(firstName, lastName) {
+    this.firstName = firstName;
+    this.lastName = lastName;
+  }
 
-const result = search(arr)
-console.log(result);
-console.log('----------');
-
-=======
-let user;
-user = 'John Doe';
-console.log(user);
-
-let student;
-// eslint-disable-next-line prefer-const
-student = 'Daria Ramanowska';
-console.log(student);
-
-user = student; //redefined data to 'Daria Ramanowska'
-console.log(user);
-
-let test = 1;
-test++;
-
-test = test + '1';
-console.log(test); // '21' string
-
-test = test - 1;
-console.log(test); // 20 number
-
-test = Boolean(test);
-console.log(test); // true
-
-console.log('----------');
-
-const arr1 = [2, 3, 5, 8];
-let res = 1;
-for (const elem of arr1) {
-  res *= elem;
+  get fullName() {
+    return `${this.firstName} ${this.lastName}`;
+  }
 }
 
-console.log(res);
-
-console.log('----------');
-
-const arr2 = [2, 5, 8, 15, 0, 6, 20, 3];
+class Student extends User {
+  constructor(firstName, lastName, admissionYear, courseName) {
+    super(firstName, lastName);
+    this.admissionYear = admissionYear;
+    this.courseName = courseName;
+  }
 
-for (const i of arr2) {
-  if (i > 5 && i < 10) {
-    console.log(i);
+  get course() {
+    const currentYear = new Date().getFullYear();
+    return currentYear - this.admissionYear;
   }
 }
-console.log('----------');
 
-for (const elem of arr2) {
-  if (elem % 3) {
-    console.log(elem);
+class Students {
+  constructor(studentList) {
+    this.list = studentList.map(
+      ({ firstName, lastName, admissionYear, courseName }) =>
+        new Student(firstName, lastName, admissionYear, courseName)
+    );
+  }
+
+  getInfo() {
+    const arr = this.list
+      .sort((a, b) => {
+        return a.course - b.course;
+      })
+      .map(
+        (student) =>
+          `${student.fullName} -  ${student.courseName} , ${student.course} курс`
+      );
+
+    console.log(arr);
```

Comment:
There is console.log() used out of the class in the example. And it is the right way because it is a side-effect here.
Just return arr and use exactly as in the example.